extension/js/common/message-renderer.ts

```diff
@@ -0,0 +1,817 @@
+      ) {
+        // there is an encrypted subject + (either there is no plain subject or the plain subject does not contain what's in the encrypted subject)
+        decryptedContent = MessageRenderer.getEncryptedSubjectText(decoded.subject, isHtml) + decryptedContent; // render encrypted subject in message
+      }
+      for (const attachment of decoded.attachments) {
+        if (attachment.isPublicKey()) {
+          publicKeys.push(attachment.getData().toUtfStr());
+        } else if (decoded.rawSignedContent && attachment.treatAs(decoded.attachments) === 'signature') {
+          signedContentInDecryptedData = { rawSignedContent: decoded.rawSignedContent, signature: attachment };
+        } else if (inlineCIDAttachments.has(attachment)) {
+          // this attachment has been processed into an inline image
+        } else {
+          renderableAttachments.push(
+            Attachment.toTransferableAttachment({
+              name: attachment.name,
+              type: attachment.type,
+              length: attachment.getData().length,
+              data: attachment.getData(),
+            })
+          );
+        }
+      }
+    }
+    if (!sigResult?.match && signedContentInDecryptedData) {
+      const plaintext = signedContentInDecryptedData.rawSignedContent;
+      const sigText = signedContentInDecryptedData.signature.getData().toUtfStr();
+      const verify = (verificationPubs: string[]) => MsgUtil.verifyDetached({ plaintext, sigText, verificationPubs });
+      const newSigResult = await verify(verificationPubs);
+      return await this.decideDecryptedContentFormattingAndRender(
+        signerEmail,
+        verificationPubs,
+        plaintext,
+        isEncrypted,
+        newSigResult,
+        renderModule,
+        this.getRetryVerification(signerEmail, verify)
+      );
+    }
+    renderModule.separateQuotedContentAndRenderText(decryptedContent, isHtml);
+    await MessageRenderer.renderPgpSignatureCheckResult(renderModule, sigResult, Boolean(signerEmail), retryVerification);
+    if (renderableAttachments.length) {
+      renderModule.renderInnerAttachments(renderableAttachments, isEncrypted);
+    }
+    renderModule.resizePgpBlockFrame();
+    return isEncrypted ? { publicKeys } : {};
+  };
+
+  private relayAndStartProcessing = async (
+    relayManager: RelayManager,
+    factory: XssSafeFactory,
+    frameId: string,
+    printMailInfo: PrintMailInfo | undefined,
+    senderEmail: string | undefined,
+    cb: (renderModule: RenderInterface, frameId: string) => Promise<{ publicKeys?: string[]; needPassphrase?: string[] }>
+  ): Promise<{ processor: Promise<unknown> }> => {
+    const renderModule = relayManager.createRelay(frameId);
+    if (printMailInfo) {
+      renderModule.setPrintMailInfo(printMailInfo);
+    }
+    const processor = cb(renderModule, frameId)
+      .then(async result => {
+        const appendAfter = $(`iframe#${frameId}`); // todo: review inbox-active-thread -- may fail
```

Comment:
Currently, `settings/inbox` rendering differs from `webmail` in a way that it first accumulates HTML strings and then adds them to the DOM. So if this is called by MessageRenderer.processAttachment, `iframe#${frameId}` may not yet exist in the DOM.
We should refactor inbox to act more like webmail, that is, first adding some placeholders to the page, and then modifying/appending to them.
Since `inbox` isn't a priority in this issue, this can be done in a separate PR.